src/typeEvaluator/typeEvaluate.ts

```diff
@@ -460,55 +457,149 @@ function handleOpCallNode(node: OpCallNode, scope: Scope): TypeNode {
 
       switch (node.op) {
-        case '==':
+        case '==': {
+          if (left.type === 'unknown' || right.type === 'unknown') {
+            return {type: 'boolean'}
+          }
+          if (left.type !== right.type) {
+            return {
+              type: 'boolean',
+              value: false,
+            } satisfies BooleanTypeNode
+          }
+          if (left.type === 'null') {
+            return {
+              type: 'boolean',
+              value: true,
+            } satisfies BooleanTypeNode
+          }
+          if (!isPrimitiveTypeNode(left) || !isPrimitiveTypeNode(right)) {
+            return {
+              type: 'boolean',
+              value: false,
+            } satisfies BooleanTypeNode
+          }
+          return {
+            type: 'boolean',
+            value: evaluateComparison(node.op, left, right),
+          } satisfies BooleanTypeNode
+        }
         case '!=': {
+          if (left.type === 'unknown' || right.type === 'unknown') {
+            return {type: 'boolean'}
+          }
+          if (left.type !== right.type) {
+            return {
+              type: 'boolean',
+              value: true,
+            } satisfies BooleanTypeNode
+          }
+          if (left.type === 'null') {
+            return {
+              type: 'boolean',
+              value: false,
+            } satisfies BooleanTypeNode
+          }
+          if (!isPrimitiveTypeNode(left) || !isPrimitiveTypeNode(right)) {
+            return {
+              type: 'boolean',
+              value: true,
+            } satisfies BooleanTypeNode
+          }
+
+          let value = evaluateComparison('==', left, right)
+          if (value !== undefined) value = !value
           return {
             type: 'boolean',
-            value: resolveCondition(node, scope),
+            value,
           } satisfies BooleanTypeNode
         }
         case '>':
         case '>=':
         case '<':
         case '<=': {
+          if (left.type === 'unknown' || right.type === 'unknown') {
+            return {type: 'unknown'}
+          }
           if (left.type !== right.type) {
+            return {type: 'null'} satisfies NullTypeNode
+          }
+          if (!isPrimitiveTypeNode(left) || !isPrimitiveTypeNode(right)) {
+            return {type: 'null'} satisfies NullTypeNode
+          }
+          return {
+            type: 'boolean',
+            value: evaluateComparison(node.op, left, right),
+          } satisfies BooleanTypeNode
+        }
+        case 'in': {
+          if (left.type === 'unknown' || right.type === 'unknown') {
+            return {type: 'unknown'}
+          }
+          if (right.type !== 'array') {
+            // Special case for global::path, since it can be used with in operator, but the type returned otherwise is a string
+            if (isFuncCall(node.right, 'global::path')) {
+              return {type: 'boolean'}
+            }
```

Comment:
A bit hacky, but just needed to short circuit the operator if used with path.

Another approach, that might come in a follow-up PR is to put the scope into `filterMode`, and we can then use that to determine what typenode to return in the function. This could also then be used to rewrite the scope when using literal comparisons or the `defined` method